urg_node/include/uam/io/tcp_client.h

```diff
@@ -168,8 +168,8 @@ class TcpClient
    *
    * @param[in] handler - Async read handler
    */
-  template <typename TReply, typename Handler>
-  void asyncReadLine(Handler handler)
+  template <typename TReply>
+  void asyncReadLine(std::function<void(const TReply&)> handler)
```

Comment:
You like the function pointer better than the template parameter? :stuck_out_tongue:

Reply:
As much as it pains me, I need to dial back a bit on the templates. 😅 
This way it becomes more obvious to the client lib what signature the handler function must have 😅